include/xrpl/beast/hash/xxhasher.h

```diff
@@ -24,29 +24,110 @@
 
 #include <xxhash.h>
 
+#include <array>
 #include <cstddef>
-#include <new>
-#include <type_traits>
+#include <cstdint>
+#include <optional>
+#include <span>
 
 namespace beast {
 
 class xxhasher
 {
+public:
+    using HashType = std::size_t;
+
 private:
     // requires 64-bit std::size_t
     static_assert(sizeof(std::size_t) == 8, "");
+    // Have an internal buffer to avoid the streaming API
+    // A 40-byte buffer should to be big enough for us
+    static constexpr std::size_t INTERNAL_BUFFER_SIZE = 40;
+
+    std::optional<XXH64_hash_t> seed_;
+    XXH3_state_t* state_ = nullptr;
 
-    XXH3_state_t* state_;
+    std::array<std::uint8_t, INTERNAL_BUFFER_SIZE> buffer_;
```

Comment:
can we make it first data member, and give it `alignas(64)` ? I would also increase `INTERNAL_BUFFER_SIZE` to 64, so the other data members start from the new cache line.